include/aca_zeta_programming.h

```diff
@@ -27,11 +27,28 @@ static atomic_uint current_available_group_id(1);
 
 namespace aca_zeta_programming
 {
+struct fwd_info {
+  string ip_addr;
+  string mac_addr;
+  fwd_info(string ip_addr = "", string mac_addr = "")
+          : ip_addr(ip_addr), mac_addr(mac_addr){};
+
+  // Overload "=="for hash operation
+  bool operator==(const fwd_info &other) const
+  {
+    if (ip_addr == other.ip_addr && mac_addr == other.mac_addr) {
```

Comment:
I would suggest simply doing this:

return ((ip_addr == other.ip_addr) && (mac_addr == other.mac_addr));